swig/include/python/gdal_python.i

```diff
@@ -4975,6 +4975,28 @@ def quiet_errors():
     finally:
         PopErrorHandler()
 
+@contextlib.contextmanager
+def quiet_warnings():
+    """Temporarily install an error handler that silences all warnings.
+
+       .. versionadded: 3.11
+
+       Returns
+       -------
+            A context manager
+
+       Example
+       -------
+
+       >>> with gdal.ExceptionMgr(useExceptions=False), gdal.quiet_errors():
```

Comment:
I assume:
```suggestion
       >>> with gdal.ExceptionMgr(useExceptions=False), gdal.quiet_warnings():
```

maybe we should have a test for that?

Reply:
I tested it manually by modifying https://github.com/OSGeo/gdal/issues/9949 . I can write an automated test (would need to involve `subprocess`, I guess?)

Reply:
(test added in 1c404c681301476ff25843e3b7a080a693812d22)